wp/theme/AbstractArchive.php

```diff
@@ -79,6 +78,28 @@ function sort_posts($a, $b){
 		return array_search($b->ID, $this->ids) > array_search($a->ID, $this->ids) ? -1 : 1;
 	}
 
+
+	function filter_frontend_main_query($allowed,$wp_query)	{
+		return $allowed && $wp_query->is_main_query() && !is_admin();
```

Comment:
what is the purpose of $allowed or adding the extra functions? seems like it would be easier to just do:

$valid = !(!$wp_query->is_main_query() || is_admin() || $this->attempted);
$valid = Config::apply_filters('should_query', $valid);